lms/static/scripts/frontend_apps/services/client-rpc.js

```diff
@@ -0,0 +1,87 @@
+import { Server, call as rpcCall } from '../../postmessage_json_rpc';
+
+/**
+ * @typedef User
+ * @prop {string} displayName
+ * @prop {string} userid
+ */
+
+/**
+ * Service for communicating with the Hypothesis client.
+ *
+ * This service handles:
+ *
+ *  - Waiting for the Hypothesis client to request its configuration and then
+ *    returning the configuration data generated by the LMS backend to the
+ *    client.
+ *  - Updating the Hypothesis client configuration in response to input
+ *    in the LMS frontend, such as changing the focused user in grading mode.
+ */
+export class ClientRpc {
+  /**
+   * Setup the RPC server used to communicate with the Hypothesis client.
+   *
+   * @param {Object} options
+   *   @param {string[]} options.allowedOrigins -
+   *     Origins that are allowed to request client configuration
+   *   @param {Object} options.clientConfig -
+   *     Configuration for the Hypothesis client. Whatever is provided here is
+   *     passed directly to the client via `window.postMessage` when it requests
+   *     configuration. It should be a subset of the config options specified at
+   *     https://h.readthedocs.io/projects/client/en/latest/publishers/config/.
+   */
+  constructor({ allowedOrigins, clientConfig }) {
+    this._server = new Server(allowedOrigins);
+
+    // Handle the initial request for configuration from the Hypothesis client.
+    this._server.register('requestConfig', () => clientConfig);
+
+    const groups = new Promise(resolve => {
+      this._resolveGroups = resolve;
+    });
+
+    // Handle the request for the list of groups to display from the Hypothesis
+    // client.
+    //
+    // Determining the list of groups to display may take time on the backend
+    // due to potentially slow calls to the external LMS to determine which course
+    // section the user is in. Therefore we separate the request for the rest of
+    // the Hypothesis client configuration from the request for the groups to
+    // show. This enables the client to do most of its startup before the
+    // group list is available.
+    this._server.register('requestGroups', () => groups);
+  }
+
+  /**
+   * Set which groups are available to select in the client.
+   *
+   * This method should be called exactly once during the LTI launch and calling
+   * it a second time will have no effect.
+   *
+   * @param {string[]} groups
+   */
+  setGroups(groups) {
+    this._resolveGroups(groups);
+  }
+
+  /**
+   * Set which user is focused in the client or none if `user` is `null`.
```

Comment:
What does `none` mean here?

Reply:
I mean that it takes the client out of "focus on a user" mode.